### PR TITLE
SASS migration for animation: declarations

### PR DESCRIPTION
**Pull request by jesstelford**
Basically just #7403, but with added support for `animate:` (and non-shorthand variants).

After diving down a rabbit hole of trying to parse `animate:` declarations, I realised the parts we care about are identical to the `transition:` ones I've already done, and the changes were minimal to add support 🎉 

Also added flags for turning on/off the transition/animation handlers (plus tests).

### CODE REVIEW

File: polaris-migrator/src/migrations/replace-sass-animatable/replace-sass-animatable.ts [hunk 1]
```diff
@@ -244,17 +261,27 @@ export default createSassMigrator(
...
-          steps: insertUnexpectedEasingFunctionComment,
-        };
+        if (node.value === namespace('easing', options)) {
+          migrateLegacySassEasingFunction(node, decl);
+          return;
+        }
 
-        if (isKeyOf(easingFuncHandlers, node.value)) {
-          easingFuncHandlers[node.value](node, decl);
+        if (ignoreUnknownFunctions) {
```
Comment (jesstelford) [inline polaris-migrator/src/migrations/replace-sass-animatable/replace-sass-animatable.ts:278]: Discovered a bug where `transition-timing-function: nonsense` was not flagged as an error. This flag (and surrounding logic catches it).

**Review by sam-b-rose — COMMENTED** [review]
Nice work incorporating `animation` declarations. I made a lot of revisions in the renaming PR (#7606) and can take over merging this into the new structure 🎉  📁 🗄️ thanks @jesstelford 🙏

**Comment by sam-b-rose** [comment]
Going to close out this PR for now. The `animation` declarations have been incorporated but still lacks some of the shorthand spec positional logic. We can reference or reopen this work if needed 👍

**Pull request closed without merging**